frontend/main.py

```diff
@@ -53,11 +62,92 @@ def aws_cognito_redirect():
                         expires=expires, httponly=True)
     return response
 
+@app.route('/corrections/<int:correction_id>', methods=['POST'])
+@aws_auth.authentication_required
+def resolve_correction(correction_id):
+    req_data = request.form
+    if 'action' not in req_data and 'corrected_hadith' not in req_data:
+        return jsonify(create_response_message("Please provide valid action param 'delete' or 'approve' and 'corrected_hadith' param"))
+
+    action = req_data['action']
+    corrected_hadith = req_data['corrected_hadith']
+    username = request.cookies.get('username')
+
+    if action == "delete":
+        return archive_correction(correction_id, username, None, False)
+
+    elif action == "approve":
+        dynamodb = boto3.resource('dynamodb',
+            endpoint_url=app.config['DYNAMODB_ENDPOINT_URL'],
+            region_name=app.config['REGION'])
+        table = dynamodb.Table(app.config['DYNAMODB_TABLE'])
+        try:
+            response = table.get_item(Key={'id': str(correction_id)})
+            if 'Item' not in response:
+                return jsonify(create_response_message("correction with id " + str(correction_id) + " not found"))
+ 
+            rows_affected = save_correction_to_hadith_table(response['Item']['urn'], corrected_hadith)
+
+            if rows_affected == 1:
+                archive_correction(correction_id, username, corrected_hadith, True)
+                return jsonify(create_response_message("Successfully updated hadith text"))
+            else:
+                return jsonify(create_response_message("Failed to update hadith text"))
+
+        except ClientError as e:
+            print(e)
+            return jsonify(create_response_message(e.response['Error']['Message']))
+        except pymysql.Error as error:
+            print(error)
+            return jsonify(create_response_message(str(error)))
+        except Exception as exception:
+            return jsonify(create_response_message("Error - " + str(exception)))
+
+    else:
+        return jsonify(create_response_message("Please provide valid action param 'delete' or 'approve'"))
 
 @app.route('/sign_in')
 def sign_in():
     return redirect(aws_auth.get_sign_in_url())
 
+def save_correction_to_hadith_table(urn, corrected_hadith):
+    conn = pymysql.connect(**mysql_properties)
+    cursor = conn.cursor()
+    query = "UPDATE bukhari_english SET hadithText = %(hadith_text)s WHERE englishURN = %(urn)s;"
```

Comment:
what is 's' next to %(hadith_text) for? Is this code tested?

Reply:
That is string formatting @hasankhan

Reply:
I'd recommend using f-strings instead of the %s notation. Also be on the lookout for SQL injection. Always sanitize input. You can use pymysql.escape_string().

Reply:
Er, don't you need quotes around the string anyway? And URN should always be an int, not a string.

Reply:
These are [prepared statements](http://zetcode.com/python/pymysql/#:~:text=PyMySQL%20prepared%20statements,statements%20increase%20security%20and%20performance.&text=In%20the%20example%2C%20we%20get%20the%20row%20with%20the%20specified%20Id.). It's not using string formatting.

Reply:
Oh good. Prepared statements are even better. I didn't know pymysql had them.

Reply:
yes I was about to say that jzk

Also for URN, dynamo db was storing it as a string so I stored it in a string as well (oops)
Ill fix that in the refactoring PR. Ill also probably also have dynamodb store things as numbers as well instead of strings

Reply:
Well. As long as the data type of URN internally consistent it's fine. Just that when you need to talk to the database, it has to be an int. And it's simpler to always keep it an int rather than remember that one detail.